### PR TITLE
docs: setup initial version of IBC-rs website with existing markdown files

### LINKED ISSUE #808: Initial setup and structure of documentation portal
## Background

As part of #743, prior to the launch of the IBC-rs documentation website, this ticket should cover:

- [ ] Set up the initial version of the portal using a static doc generator framework, laying the foundation for further additions to be made on a feature branch.
- [ ] Show existing content/markdowns as the initial resource
- [ ] Organize the documentation structure and create sections and subsections
- [ ] Provide sidebar navigation for easy access
- [ ] Ensure consistent formatting and styling throughout the portal
- [ ] Add a CI job for checking documentation build

### PR DESCRIPTION
**Pull request by Farhad-Shabani**
Closes: #808

### This PR:

- Generates site by Markdown files using Docusaurus framework
- Uses the theme from Cosmos-SDK with a few adjustments
- Splits out the `Releases` section from `CONTRIBUTING.md`
- During deployment, copies certain markdown files (`/architecture`, `CHANGELOG.md`, and `CONTRIBUTING.md`) to the `/docs` directory, but the rest of markdowns are supposed to live in `docs`.
- Allows navigation from the navbar through two top-level categories, `learn` and `developers` with sections for ADRs, `CHANGELOG`, and `CONTRIBUTING`
- Uses the `Edit this page` link for each section
- In a general view, markdowns under the docs are organized as follows:
```markdown
docs/learn/
├── 01-intro
│   │ ...
├── 03-clients
│    └── 01-tendermint.md
└── 04-apps
     ├── 01-token-transfer.md
     └── 02-interchain-accounts.md

docs/developers/
├── 00-intro
├── 01-integration
│    └── 01-integrate-ibc-rs.md
│    ├── 02-integrate-grpc.md
│    └── 03-integrate-hermes.md
├── 02-upgrades
├── 03-clients
├── 04-apps
├── 05-architecture
└── 06-migrations
```
______

### PR author checklist:
- [ ] Added changelog entry, using [`unclog`](https://github.com/informalsystems/unclog).
- [ ] Added tests.
- [x] Linked to GitHub issue.
- [ ] Updated code comments and documentation (e.g., `docs/`).
- [ ] Tagged *one* reviewer who will be the one responsible for shepherding this PR.

### Reviewer checklist:

- [ ] Reviewed `Files changed` in the GitHub PR explorer.
- [ ] Manually tested (in case integration/unit/mock tests are absent).

### CODE REVIEW

File: RELEAESE.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+## Releases
```
Comment (plafer) [inline RELEAESE.md]: There's a typo in the file name (should me `RELEASE.md`)

File: docs/DOC_README.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+updates to files in the `/docs` directory will automatically trigger a website
+deployment.
+
+## How to Build the Docs Locally
+
+Go to the `docs` directory and run the following commands:
+
+```shell
+cd docs
+npm install
```
Comment (plafer) [inline docs/DOC_README.md:50]: what version of `node` are we using?

File: docs/docs/learn/01-intro/00-intro.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+IBC-rs is designed to serve the complete suite of IBC protocol standards. As
+such, it supports all the core specifications essential for TAO (Transfer,
+Authentication, Ordering), spanning from ICS-02 to ICS-05, as well as ICS-24 and
+ICS-26. However, a few
+[divergences](https://github.com/cosmos/ibc-rs/tree/main/crates/ibc#divergence-from-the-interchain-standards-ics)
+from the interchain standards (ICS) are introduced to accommodate the
+implementation.
+
+* **IBC Light Clients**
+  * [**ICS-20: Tendermint**](./../learn/clients/tendermint)
```
Comment (plafer) [inline docs/docs/learn/01-intro/00-intro.md]:
```suggestion
  * [**ICS-7: Tendermint**](./../learn/clients/tendermint)
```

File: docs/docs/learn/01-intro/00-intro.md [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+  architectural design decisions.
+* [**FAQ Page**](https://github.com/cosmos/ibc-rs/wiki/FAQ) - Browse frequently asked
+  questions about IBC-rs.
+
+## How to Integrate IBC
+
+* [**Integration**](./../../developers/integration/overview) - Learn how to
+  integrate IBC module and relayer into your blockchain.
+
+## Explore the Stack
```
Comment (plafer) [inline docs/docs/learn/01-intro/00-intro.md]:
```suggestion
## Related projects
```
Moving away from "Stack" terminology, as related projects don't form a stack

File: docs/docs/learn/01-intro/00-intro.md [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+
+Check out the docs of projects relate to the IBC-rs stack:
```
Comment (plafer) [inline docs/docs/learn/01-intro/00-intro.md]:
```suggestion
Check out the docs of projects related to IBC-rs:
```
Similarly, moving away from "stack"; ibc-rs would be part of a stack, but itself isn't a stack

File: docs/docs/learn/01-intro/00-intro.md [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+  implementation of the CometBFT ABCI application, primarily serving as our
+  testing stand.
+* [**IBC-Go**](https://ibc.cosmos.network/) - A Go implementation of the IBC
+  protocol.
+
+## Security Notice
+
+There are no established security practices specifically for IBC-rs. Please
+direct your security reports to the [IBC-go
+repository](https://github.com/cosmos/ibc-go/security/policy).
```
Comment (plafer) [inline docs/docs/learn/01-intro/00-intro.md]: This is not right; there can be vulnerabilities which are specific to our implementation (and have nothing to do with ibc-go). If someone finds one, they should use the Github feature to report vulnerabilities to the repo (and we should enable it if it needs to be enabled).

File: docs/docs/learn/01-intro/00-intro.md [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+
+**Note**: The [disclosure
+  log](https://github.com/informalsystems/hermes/blob/master/docs/disclosure-log.md) documents
+  problems we have uncovered while specifying and implementing IBC-rs. Some of
+  the cases recorded there might have been fixed.
```
Comment (plafer) [inline docs/docs/learn/01-intro/00-intro.md]: This log refers to issues we found at a protocol level. While some *could* be argued to be security-related, I find it more confusing than anything to put this in the front page of the ibc-rs website.

Essentially IMO the security notice here should only tell users that if they find a bug, please file a security advisory on the repo, and we'll fix it quickly.

**Review by plafer — CHANGES_REQUESTED** [review]
Awesome work! Left a few minor things to change, after which we're good to merge

**Comment by plafer** [comment]
Also please add `node_modules/` and `.docusaurus/` to the `.gitignore`

File: docs/docs/learn/01-intro/00-intro.md (hunk 6 above)
Comment (Farhad-Shabani) [reply docs/docs/learn/01-intro/00-intro.md]: I see. Opened https://github.com/cosmos/ibc-rs/issues/824 so separately have it reviewed and discussed later.

File: docs/docs/learn/01-intro/00-intro.md (hunk 7 above)
Comment (Farhad-Shabani) [reply docs/docs/learn/01-intro/00-intro.md]: Right. dropped.

File: docs/DOC_README.md (hunk 2 above)
Comment (Farhad-Shabani) [reply docs/DOC_README.md:50]: Node >= 16.x and npm >= 8.5.
Added this line above.

**Review by plafer — APPROVED** [review]
🙏

**Pull request merged**